include/psa/crypto_struct.h

```diff
@@ -223,9 +223,34 @@ static inline struct psa_key_derivation_s psa_key_derivation_operation_init(
     return v;
 }
 
-struct psa_key_production_parameters_s {
+struct psa_custom_key_parameters_s {
     /* Future versions may add other fields in this structure. */
     uint32_t flags;
+};
+
+/** The default production parameters for key generation or key derivation.
+ *
+ * Calling psa_generate_key_custom() or psa_key_derivation_output_key_custom()
+ * with `custom=PSA_CUSTOM_KEY_PARAMETERS_INIT` and `custom_data_length=0` is
+ * equivalent to calling psa_generate_key() or psa_key_derivation_output_key()
+ * respectively.
+ */
+#define PSA_CUSTOM_KEY_PARAMETERS_INIT { 0 }
+
+#ifndef __cplusplus
+/* Omitted when compiling in C++, because one of the parameters is a
+ * pointer to a struct with a flexible array member, and that is not
+ * standard C++.
+ * https://github.com/Mbed-TLS/mbedtls/issues/9020
+ */
+/* This is a deprecated variant of `struct psa_custom_key_parameters_s`.
+ * It has exactly the same layout, plus an extra field which is a flexible
+ * array members. Thus a `const struct psa_key_production_parameters_s*`
```

Comment:
(minor) should be `array member` (singular)

(match code style) ``const struct psa_key_production_parameters_s *` (with space before `*`, also on line 250)

Reply:
I don't understand the “match code style” part (it doesn't help that the formatting on GitHub is wrong).

Reply:
This is about there not being a space between `psa_key_production_parameters_s` and `*`, which is what we do in code (in code there is a space before the `*`, e.g.

```
const struct psa_key_production_parameters_s *params;
```

so in documentation I would expect to see

```
Thus a `const struct psa_key_production_parameters_s *`
```

Reply:
LGTM but agree with this comment